core/translate/main_loop.rs

```diff
@@ -442,6 +442,23 @@ pub fn init_loop(
                 }
                 _ => panic!("only SELECT is supported for index method"),
             },
+            Operation::HashJoin(_) => {
+                match mode {
+                    OperationMode::SELECT => {
+                        // Open probe table cursor, the build table cursor should already be open from a previous iteration.
```

Comment:
error if it isn't? looks like we're silently ignoring the case where `table_cursor_id` is None or `&table.table` is not a btree